addons/hr_timesheet/models/project_task.py

```diff
@@ -174,6 +174,8 @@ def action_view_subtask_timesheet(self):
         graph_view_id = self.env.ref("hr_timesheet.view_hr_timesheet_line_graph_by_employee").id
         new_views = []
         for view in action['views']:
+            if self.env.context.get('is_project_sharing', False) and view[1] not in ['tree', 'kanban', 'form']:
```

Comment:
```suggestion
            if (not is_internal_user or self.env.context.get('is_project_sharing', False)) and view[1] not in ['tree', 'kanban', 'form']:
```

The context key will be there is the module is updated but if it is not we could remove the view other than tree, kanban and form when the user is not an internal user. 🙂